TLSharp.Core/Network/TcpTransport.cs

```diff
@@ -1,7 +1,8 @@
 ﻿using System;
-using System.Linq;
+using System.Diagnostics;
 using System.Net;
 using System.Net.Sockets;
+using System.Reflection;
```

Comment:
Ok, seems like the original contributor used this: http://stackoverflow.com/questions/17751835/connecting-to-a-tcp-server-via-tcpclient-through-proxy

However, using Reflection this way is very likely to cause it to not work with Mono (e.g. in Linux/Mac), so I would prefer if you implement it this way: http://stackoverflow.com/questions/18289418/c-sharp-tcpclient-connect-via-a-proxy